go-controller/pkg/node/gateway_localnet.go

```diff
@@ -138,18 +140,23 @@ func (npw *localPortWatcherData) addService(svc *kapi.Service) error {
 				klog.V(5).Infof("ExternalIP: %s is reachable through one of the interfaces on this node, will skip setup", externalIP)
 			} else {
 				if gatewayIP != "" {
-					if stdout, stderr, err := util.RunIP("route", "replace", externalIP, "via", gatewayIP, "dev", localGwModeNextHopPort, "table", localnetGatewayExternalIDTable); err != nil {
-						klog.Errorf("Error adding routing table entry for ExternalIP %s, via gw: %s: stdout: %s, stderr: %s, err: %v", externalIP, gatewayIP, stdout, stderr, err)
-					} else {
-						klog.V(5).Infof("Successfully added route for ExternalIP: %s", externalIP)
-					}
+					routeUsage[externalIP] = true
 				} else {
 					klog.Warningf("No gateway of appropriate IP family for ExternalIP %s for Service %s/%s",
 						externalIP, svc.Namespace, svc.Name)
 				}
 			}
 		}
 	}
+	for externalIP, needsRoute := range routeUsage {
+		if needsRoute {
```

Comment:
When will needsRoute ever be false? Just do `for externalIP, _ := range routeUsage` or something like that.

Reply:
done